### PR TITLE
Fix: modal issues

### PR DESCRIPTION
**Pull request by abhishekmmgn**
**Description**

This PR fixes #

**Notes for Reviewers**
List of changes/fixes:

Added spacing between filters and the catalog
z index and top margin of the help modal.
Read more shows only if there is more content
General spacing
General UI issues
Added mini card (related pattern card) on viewport width ~600 - 1150
changed ...read more to Read more
Added button to meshery.io/extensions card
Converted title to link and changed external link icon to internal
Changed the service mesh-looking image to a service-mesh-pattern image 
Add related patterns on small-width
The gap in the catalog

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by abhishekmmgn** [comment]
@yash37158

**Comment by yash37158** [comment]
@abhishekmmgn Can you please point to the issue you were working on ?

**Comment by yash37158** [comment]
image

The top spacing for the "Extension" title is not responsive. It was working correctly previously.

**Comment by ritiksaxena124** [comment]
Hey @abhishekmmgn  let's discuss this on today's [website meet]([https://meet.layer5.io/websites](https://www.google.com/url?q=http://meet.layer5.io/websites&sa=D&source=calendar&usd=2&usg=AOvVaw1CNpNwmYd_rTFcQa_7fBB5)) if you are available. I have added it as an [agenda item](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit)

**Comment by abhishekmmgn** [comment]
We can merge the PR now if it can. There are a lot of commits on top of it now, so I get merge conflicts when pushing the changes.
In next commit, I'll remove the buttons in the extensions catalog
And will fix the h2 heading in extensions and also in the modals.

**Comment by leecalcote** [comment]
@iArchitSharma review, please.

**Review by iArchitSharma — APPROVED** [review]
LGTM!

File: _includes/modal.html [hunk 1]
```diff
@@ -1,76 +1,138 @@
...
+      />
+      {%else%}
+      <img
+        class="image-class"
         data-snapshot-light="https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/{{pattern.patternId}}-light.png"
         data-snapshot-dark="https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/{{pattern.patternId}}-dark.png"
         src="https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/{{pattern.patternId}}-dark.png"
-        onerror="this.src='/assets/images/patterns/service-mesh.svg'" loading="lazy" onclick="toggleFullScreenImage('{{ pattern.patternId }}')" />
-        {%endif%}
+        onerror="this.src='{{site.baseurl}}/assets/images/logos/service-mesh-pattern.svg', this.style.width = '50%'"
```
Comment (vishalvivekm) [inline _includes/modal.html:33]: fixes #1744


cc: #1755 #1746

**Pull request merged**